### PR TITLE
refactors statsmodels and  theta forecaster

### PR DESCRIPTION
**Pull request by thayeylolu**
Reference : https://github.com/alan-turing-institute/sktime/issues/955

We need to slowly re-factor all forecasters to be compliant with the new interface (see PR #912). i.e., no overrides of fit etc and descendant classes only implementing core logic, while the BaseForecaster implements check etc logic.

Currently this is running due to various downwards compatibility workarounds, which should be removed once all forecasters are ready (in "no-override" state)

The forecasters that still need a downstream refactor are:

  

- [x] _StatsModelAdapter and descendants AutoETS, ExponentialSmoothing, ThetaForecaster

### CODE REVIEW

File: sktime/forecasting/base/adapters/_statsmodels.py [hunk 1]
```diff
@@ -41,11 +45,7 @@ def fit(self, y, X=None, fh=None):
         # so we coerce them here to pd.RangeIndex
         if isinstance(y, pd.Series) and type(y.index) == pd.Int64Index:
```
Comment (fkiraly) [inline sktime/forecasting/base/adapters/_statsmodels.py:46]: hm, this should perhaps be a `BaseForecaster` functionality - let's make a note of this (but no action required)

**Comment by thayeylolu** [comment]
@mloning this is ready for review :)

**Review by fkiraly — APPROVED** [review]
looks good to me, we need to update from main and let the tests pass

**Comment by fkiraly** [comment]
tests passed overnight - tests have re-started due to a PR that only fixed a broken link, so force-merging

**Pull request merged**